File: sem2_c_cpp_java/05-04-23/SampleCodes_InnerClassObjectsCPP_Java/1.cpp
#include<iostream>

using namespace std;

class A1
{
	public:
		void f1()
		{
			cout << "Hello from f1 of A1" << endl;
		}

		class B1
		{
			public:
				void f2()
				{
					cout << "Hello from f2 of B1" << endl;
				}
		};
};

int main()
{
	class A1 a1;
	a1.f1();
	class A1::B1 b1;
	b1.f2();
}

File: sem2_c_cpp_java/05-04-23/SampleCodes_InnerClassObjectsCPP_Java/1.java
import java.util.*;

class A1
{
	public void f1()
	{
		System.out.println("Hello from f1() of A1");
	}

	class B1
	{
		public void f2()
		{
			System.out.println("Hello from f2() of A2");
		}
	};

};

class C1
{
	public static void main(String args[])
	{
		A1 a1 = new A1();
		a1.f1();
		A1.B1 b1 = new A1().new B1();
		b1.f2();
		b1.f1();
	}
};
